packages/core/ios/EmbraceManager.swift

```diff
@@ -21,83 +22,98 @@ private let EVENT_ATTRIBUTES_KEY = "attributes"
+        DispatchQueue.global(qos: .background).async {
+            do {
+                let bundleID = try computeBundleID(path: path)
+                try Embrace.client?.metadata.addResource(key: REACT_NATIVE_BUNDLE_ID_RESOURCE_KEY, value: bundleID.id, lifespan: .process)
+                resolve(true)
+            } catch let error {
+                reject("SET_JS_BUNDLE_PATH_ERROR", "Error setting JavaScript bundle path", error)
+            }
         }
     }
-  }
+
     @objc
     func getDefaultJavaScriptBundlePath(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        if  let filePath = Bundle.main.path(forResource: "main", ofType: "jsbundle"){
+        if  let filePath = Bundle.main.path(forResource: "main", ofType: "jsbundle") {
             resolve(filePath)
         } else {
             reject("error", "Unable to retrieve JS bundle path", nil)
         }
     }
+
     @objc
     func isStarted(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
         if let embraceStarted = Embrace.client?.started {
-        resolve(embraceStarted)
+            resolve(embraceStarted)
         } else {
-        resolve(false)
+            resolve(false)
         }
     }
 
     @objc(startNativeEmbraceSDK:resolver:rejecter:)
     func startNativeEmbraceSDK(configDict: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        let config = SDKConfig(from: configDict)
+        config = SDKConfig(from: configDict)
+
         DispatchQueue.main.async {
             do {
                 var embraceOptions: Embrace.Options {
                     var crashReporter: CrashReporter?
-                    if config.disableCrashReporter {
-                        crashReporter = nil
-                    } else {
-                        crashReporter = EmbraceCrashReporter()
+                    crashReporter = self.config.disableCrashReporter ? nil : EmbraceCrashReporter()
+
+                    let servicesBuilder = CaptureServiceBuilder()
+
+                    if self.config.disableNetworkSpanForwarding {
+                        // allowing to disable NSF by code
+                        let urlSessionServiceOptions = URLSessionCaptureService.Options(injectTracingHeader: false, requestsDataSource: nil)
+
+                        // manually adding the URLSessionCaptureService
+                        servicesBuilder.add(.urlSession(options: urlSessionServiceOptions))
                     }
 
-                    let servicesBuilder = CaptureServiceBuilder().addDefaults()
-                    if config.disableAutomaticViewCapture {
-                            servicesBuilder.remove(ofType: ViewCaptureService.self)
+                    // adding defaults
+                    servicesBuilder.addDefaults()
```

Comment:
hmm isn't this still adding the urlSessionService again though if `self.config.disableNetworkSpanForwarding` was set? Or is `addDefaults` checking that it doesn't re-add a service that is already on the builder?

Reply:
as per what I understood from the message I shared before the service won't be replaced unless I'm wrong. @ArielDemarco @NachoEmbrace could you please take a look? is this the right way of implementing this cenario? is there any improvement I can work on?

Reply:
If you already set a specific `CaptureService` (like `URLSessionCaptureService`) the `addDefaults` method won't override it. This method checks if the service was already added.